sdk/nodejs/cmd/pulumi-language-nodejs/main_test.go

```diff
@@ -408,31 +408,48 @@ func TestNonblockingStdout(t *testing.T) {
 os.set_blocking(1, False) # set stdout to non-blocking
 time.sleep(3)
 `
+
+	// Create a named pipe to use as stdout
+	tmp := os.TempDir()
+	p := filepath.Join(tmp, "fake-stdout")
+	err := syscall.Mkfifo(p, 0o644)
+	defer os.Remove(p)
+	require.NoError(t, err)
+	// Open fd without O_NONBLOCK, ensuring that os.NewFile does not return a pollable file.
+	// When our python script changes the file to non-blocking, Go does not notice and continues to
+	// expect the file to be blocking, and we can trigger the bug.
```

Comment:
Is it worth a line or two doc-commenting what the bug is above the test function? The issue linked above hints that for some reason we think `stdout` is unavailable, but it's not super clear (to me at least) why that is?

Reply:
Ah apologies for missing that. Thanks! :pray: